src/khoj/routers/api.py

```diff
@@ -341,24 +320,26 @@ async def set_processor_enable_offline_chat_config_data(
         offline_chat_model: Optional[str] = None,
         client: Optional[str] = None,
     ):
-        _initialize_config()
+        user = request.user.object
 
-        if not state.config.processor or not state.config.processor.conversation:
-            default_config = constants.default_config
-            default_conversation_logfile = resolve_absolute_path(
-                default_config["processor"]["conversation"]["conversation-logfile"]  # type: ignore
+        if enable_offline_chat:
+            conversation_config = ConversationProcessorConfig(
+                offline_chat=OfflineChatProcessorConfig(
+                    enable_offline_chat=enable_offline_chat,
+                    chat_model=offline_chat_model,
+                )
             )
-            conversation_logfile = resolve_absolute_path(default_conversation_logfile)
-            state.config.processor = ProcessorConfig(conversation=ConversationProcessorConfig(conversation_logfile=conversation_logfile))  # type: ignore
 
-        assert state.config.processor.conversation is not None
-        if state.config.processor.conversation.offline_chat is None:
-            state.config.processor.conversation.offline_chat = OfflineChatProcessorConfig()
+            await sync_to_async(ConversationAdapters.set_conversation_processor_config)(user, conversation_config)
 
-        state.config.processor.conversation.offline_chat.enable_offline_chat = enable_offline_chat
-        if offline_chat_model is not None:
-            state.config.processor.conversation.offline_chat.chat_model = offline_chat_model
-        state.processor_config = configure_processor(state.config.processor, state.processor_config)
+            offline_chat = await ConversationAdapters.get_offline_chat(user)
+            chat_model = offline_chat.chat_model
+            if state.gpt4all_processor_config is None:
+                state.gpt4all_processor_config = GPT4AllProcessorModel(chat_model=chat_model)
+
+        else:
+            await sync_to_async(ConversationAdapters.clear_offline_chat_conversation_config)(user)
+            state.gpt4all_processor_config = None
```

Comment:
This should be a user specific state variable now?

Reply:
In our self-hosted model, we can't really support multiple offline models on the server. May need to think more about how to make this apparent in the UX, but it makes sense for it to be shared across the application state when self-hosted.